src/bin/rusty_usn.rs

```diff
@@ -0,0 +1,182 @@
+#[macro_use]
+extern crate log;
+extern crate clap;
+extern crate chrono;
+use std::fs;
+use std::path::Path;
+use log::LevelFilter;
+use std::process::exit;
+use clap::{App, Arg, ArgMatches};
+use RustyUsn::usn::{UsnParserSettings, UsnParser};
+
+static VERSION: &'static str = "1.0.0";
+
+
+fn is_a_non_negative_number(value: String) -> Result<(), String> {
+    match value.parse::<usize>() {
+        Ok(_) => Ok(()),
+        Err(_) => Err("Expected value to be a positive number.".to_owned()),
+    }
+}
+
+
+fn make_app<'a, 'b>() -> App<'a, 'b> {
+    let source_arg = Arg::with_name("source")
+        .short("s")
+        .long("source")
+        .value_name("PATH")
+        .help("The source to parse.")
+        .takes_value(true);
+
+    let thread_count = Arg::with_name("threads")
+        .short("-t")
+        .long("--threads")
+        .default_value("0")
+        .validator(is_a_non_negative_number)
+        .help("Sets the number of worker threads, defaults to number of CPU cores.");
+
+    let verbose = Arg::with_name("debug")
+        .short("-d")
+        .long("debug")
+        .value_name("DEBUG")
+        .takes_value(true)
+        .possible_values(&["Off", "Error", "Warn", "Info", "Debug", "Trace"])
+        .help("Debug level to use.");
+
+    App::new("rusty_usn")
+        .version(VERSION)
+        .author("Matthew Seyer <https://github.com/forensicmatt/RustyUsn>")
+        .about("USN Parser written in Rust.")
+        .arg(source_arg)
+        .arg(thread_count)
+        .arg(verbose)
+}
+
+
+fn set_debug_level(matches: &ArgMatches){
+    // Get the possible logging level supplied by the user
+    let message_level = match matches.is_present("debug") {
```

Comment:
Personally I think `-v, -vv, -vvv` is nicer (and more common), but it's a matter of taste :)

Reply:
I will leave this. I like being able to have the one to one with the exact verbage of the logging lib.